feature: create bookmark

Woogie95 opened the pull request:
## Changes
- 북마크 추가 API 를 추가하였습니다.
- Location /api/bookmarks/:bookmarkId  Header 에 담아서 반환하도록 하였습니다.

## Background
- 관심있는 입양 게시물을 추가하여 따로 보기 위하여 API 를 추가

## Issues
#95 

## Execute
- [x] 포스트맨 테스트 
- [x] 테스트 코드
<img width="600" alt="포스트맨_테스트" src="https://github.com/banzzokee/banzzokee-back/assets/78896254/fef68a87-35ed-4879-a4fb-ab1222d6c44a">
<img width="600" alt="북마크_등록_컨트롤러_테스트" src="https://github.com/banzzokee/banzzokee-back/assets/78896254/00b81805-a14f-4219-94a9-9da89c33d0bf">
<img width="600" alt="북마크_등록_서비스_테스트" src="https://github.com/banzzokee/banzzokee-back/assets/78896254/0df23f3f-2eba-4449-836a-821951910910">

Woogie95 committed:
chore: JavaMailSender 에러 발생으로 MailConfig 추가 & SMTP yml 이름 변경

Woogie95 committed:
refactor: UserDetailsImpl & UserDetailsServiceImpl 권한 처리

- 하는 부분 Set<Role> 타입으로 되어있어서 List -> Set 으로 변경

Woogie95 committed:
test: 북마크 등록 테스트 코드 추가

Woogie95 committed:
chore: 어노테이션 & 코드 라인 정리

- @Builder 추가
- UserDetailsImpl Set -> List 로 변경

Woogie95 committed:
test: 테스트 코드 MockUser 관련 코드 추가

Woogie95 committed:
test: Java CI with Gradle fail

- USER -> ROLE_USER

nadoran78 commented on src/main/java/homes/banzzokee/domain/bookmark/controller/BookmarkController.java:
```diff
@@ -0,0 +1,38 @@
+    bookmarkService.registerBookmark(userDetails.getUserId(), bookmarkRegisterRequest.getAdoptionId());
+    URI location = ServletUriComponentsBuilder.fromCurrentRequest()
+        .path("/{bookmarkId}")
+        .buildAndExpand(bookmarkRegisterRequest.getAdoptionId()).toUri();
```
bookmarkId 자리에 adoptionId가 입력되는 거 같은데.. 이게 이해가 잘 안 됩니다.

eTaphee replied:
adoptionId가 아니고 bookmarkId 아닐까요?

Woogie95 replied:
```
.path("/{adoptionId}")
        .buildAndExpand(bookmarkRegisterRequest.getAdoptionId()).toUri();
```        
이렇게 수정 했는데 adoptionId 를 북마크에 추가하는 하는 방향이니까 이렇게 아닌가요???

Woogie95 committed:
chore: 북마크 컨트롤러 넘겨주는 값 수정

Woogie95 committed:
chore: elasticsearch.yml 환경 변수 수정

- ${ES_FINGERPRINT}

Woogie95 committed:
chore: UserDetailsServiceImpl 권한 넘겨주는 부분 수정

- 변경 전 "USER_" + role.name()
- 변경 후 role.name() enum 클래스에 USER_ 정의 하였기 때문에 Impl 클래스에서 제거

Woogie95 committed:
test: 북마크 컨트롤러 & 서비스 테스트 수정

Woogie95 committed:
chore: registerBookmark 경로 변수명 수정

- bookmarkId -> adoptionId

Woogie95 committed:
chore: git commit 충돌 해결

Woogie95 committed:
chore: git commit 충돌 해결

Woogie95 committed:
refactor: git pull Request 시 빌드 실패로 AdoptionController 수정

- userDetails.getUser().getId()); -> userDetails.getUserId());

Woogie95 committed:
test: git pull Request 시 빌드 실패로 AdoptionControllerTest 수정

- when(userDetails.getUser()).thenReturn(user);
 -> when(userDetails.getUserId()).thenReturn(1L);
